content/posts/gitops-series-part-1.md

```diff
@@ -0,0 +1,616 @@
+
+We're going to create a `cluster.tf` file.
+
+It's going to look like this: `terraform/cluster.tf`
+
+```terraform
+provider "google" {
+  project = "${var.project}"
+  region  = "${var.region}"
+}
+
+# Change this section
+terraform {
+  backend "gcs" {
+    bucket  = "my-unique-bucket"
+    prefix  = "terraform/state"
+    project = "my-project"
+  }
+}
+
+resource "google_container_cluster" "cluster" {
+  name               = "${var.cluster_name}"
+  location           = "${var.cluster_zone}"
+  min_master_version = "${var.cluster_k8s_version}"
+
+  addons_config {
+    network_policy_config {
+      disabled = true
+    }
+
+    http_load_balancing {
+      disabled = false
+    }
+
+    kubernetes_dashboard {
+      disabled = false
+    }
+  }
+
+  node_pool {
+    name               = "default-pool"
+    initial_node_count = "${var.initial_node_count}"
+
+    management {
+      auto_repair = true
+    }
+
+    autoscaling {
+      min_node_count = "${var.autoscaling_min_node_count}"
+      max_node_count = "${var.autoscaling_max_node_count}"
+    }
+
+    node_config {
+      preemptible  = false
+      disk_size_gb = "${var.disk_size_gb}"
+      disk_type    = "${var.disk_type}"
+
+      machine_type = "${var.machine_type}"
+
+      oauth_scopes = [
+        "https://www.googleapis.com/auth/devstorage.read_only",
+        "https://www.googleapis.com/auth/logging.write",
+        "https://www.googleapis.com/auth/monitoring",
+        "https://www.googleapis.com/auth/service.management.readonly",
+        "https://www.googleapis.com/auth/servicecontrol",
+        "https://www.googleapis.com/auth/trace.append",
+        "https://www.googleapis.com/auth/compute",
+        "https://www.googleapis.com/auth/cloud-platform"
+      ]
+
+    }
+  }
+}
+
+output "client_certificate" {
+  value     = "${google_container_cluster.cluster.master_auth.0.client_certificate}"
+  sensitive = true
+}
+
+output "client_key" {
+  value     = "${google_container_cluster.cluster.master_auth.0.client_key}"
+  sensitive = true
+}
+
+output "cluster_ca_certificate" {
+  value     = "${google_container_cluster.cluster.master_auth.0.cluster_ca_certificate}"
+  sensitive = true
+}
+
+output "host" {
+  value     = "${google_container_cluster.cluster.endpoint}"
+  sensitive = true
+}
+```
+
+This may look complicated, but really all we are doing is defining the configuration of the cluster we want to provision. As you can see we are taking full use of the variables we listed in the `variables.tf` file.
+
+There is one section you will need to modify, it is the following block:
+
+```terraform
+terraform {
+  backend "gcs" {
+    bucket  = "my-unique-bucket"
+    prefix  = "terraform/state"
+    project = "my-project"
+  }
+}
+```
+
+If you're following along, you can follow these [instructions](https://cloud.google.com/storage/docs/creating-buckets) to create a GCS bucket here.
```

Comment:
You may want to explain what the bucket will be used for - or did I miss it?

Reply:
Good call, forgot to write that chunk